Drop compatibility for anything less than v2.092 & xenial

Geod24 opened the pull request:
```
Since we want to drop support for dmd-transitional,
we need to settle on the minimum supported version of DMD.
Due to the various deprecations and some codeggen bugs in -O -inline,
only v2.092.0 and higher is able to compile our code correctly.

Additionally we now only test on bionic instead of xenial,
and reduced coverage to only the latest version of the compiler.
```

Geod24 committed:
Drop compatibility for anything less than v2.092

Since we want to drop support for dmd-transitional,
we need to settle on the minimum supported version of DMD.
Due to the various deprecations and some codeggen bugs in -O -inline,
only v2.092.0 and higher is able to compile our code correctly.

Additionally we now only test on bionic instead of xenial,
and reduced coverage to only the latest version of the compiler.

ibuclaw commented:
Can we add [focal](https://docs.travis-ci.com/user/reference/focal/)?

Geod24 commented:
Well the distribution we run on is dictated by the docker image, it doesn't actually run on Travis, but in a docker container running on Travis. Perhaps we could just simplify it all the way ?

ibuclaw reviewed (changes requested):
Computer says no.

ibuclaw commented:
> Well the distribution we run on is dictated by the docker image, it doesn't actually run on Travis, but in a docker container running on Travis. Perhaps we could just simplify it all the way ?

We can build an deploy a [new docker image then](https://github.com/sociomantic-tsunami/cachalot/blob/v7.x.x/Makefile#L10)?

Dropping cachalot would need to be a decision made by the SC.

Geod24 commented:
> Dropping cachalot would need to be a decision made by the SC.

Agreed, which is why I don't want to conflate adding support for new images which were never used at Sociomantic with removing `dmd-transitional` (a decision already agreed on).  So if you don't mind, support for Focal will be in a future (™️ ) PR.

Geod24 committed:
Remove stomping prevention code

This requires a specific druntime which is package with dmd-transitional.
While useful, it is the only feature preventing dropping this custom compiler.
